arcwise/agent/execute_sql.py

```diff
@@ -214,6 +223,35 @@ def _check_tables_used_for_division(node: exp.Div, sg_query: exp.Query):
 
 Please rework the query and try again.
 """)
+    
+def _check_output_types(
+    columns: list[str],
+    rows: list[list[SQLScalar]],
+    predicted_output_types: list[SchemaPredictions.OutputType],
+):
+    if len(rows) == 0:
+        return
+
+    if len(columns) != len(predicted_output_types):
+        raise RuntimeError(
+            f"Expected {len(predicted_output_types)} columns, got {len(columns)}"
+        )
+
+    # check whether the types match in the first row
+    for val, pred in zip(rows[0], predicted_output_types):
+        if val is None:
+            # None values can be anything
+            continue
+        elif pred.type in {"text", "date", "datetime"} and isinstance(val, str):
+            continue
+        elif pred.type == "integer" and isinstance(val, int):
```

Comment:
Will have to check the stats later, but I wonder if it’s best to give some room for error and allow mixing of integers and reals (rarely the model mixes the two.)

Reply:
OK just checked and there’s 4/500 examples in the minidev set where the prediction is off wrt integer vs real, so not sure if it matters (maybe it’s fine anyways if we only check the cardinality anyhow?)

Reply:
Mentioned in [another comment](https://github.com/arcwisedata/bird-sql/pull/2#discussion_r1685828445), but I had seen at least one instance where checking more than just the cardinality was helpful, so I feel it's worth keeping